translate docs/faq.md (#131)
Nextjs-ja-translation/Nextjs-ja-translation-docs: docs/faq.md

> description: Get to know more about Next.js with the frequently asked questions.

Reviewer: ```suggestion
description: よくある質問を通して Next.js について理解を深めましょう。
```

description 部分の訳を追加しました。

---

# よくある質問

<details>
  <summary>どのブラウザに対応していますか？</summary>

>   <p>Next.js は <a href="https://new.babeljs.io/docs/en/next/babel-preset-env.html">@babel/preset-env</a> を利用することで IE11 を含む現行の全てのブラウザに対応しています。また IE11 に対応するため、 Next.js は Promise polyfill をグローバルインストールしています。</p>

Reviewer 2: ```suggestion
  <p>Next.js は <a href="https://new.babeljs.io/docs/en/next/babel-preset-env.html">@babel/preset-env</a> を利用することで IE11 とすべてのモダンブラウザに対応しています。また IE11 に対応するため、 Next.js は Promise polyfill をグローバルインストールしています。</p>
```
`IE11 and all modern browsers` とあるのですべてのブラウザではなく、 IE11(レガシーブラウザ) + モダンブラウザに対応している訳が適切かなと思いました

>   <p>目的のブラウザが対応していない機能を必要とするコードや外部 npm パッケージを使用する場合は、 polyfills を実装する必要があります。 polyfills を実装する必要がある場合、 <a href="https://github.com/zeit/next.js/tree/canary/examples/with-polyfills">polyfills</a> のサンプルで推奨されている方法を見ることができます。</p>

Reviewer 2: ```suggestion
  <p>目的のブラウザが対応していない機能を必要とするコードや外部 npm パッケージを使用する場合は、 Polyfill を実装する必要があります。 Polyfill を実装する必要がある場合、 <a href="https://github.com/vercel/next.js/tree/canary/examples/with-polyfills">Polyfill</a> のサンプルで推奨されている方法を見ることができます。</p>
```
日本語で用いられる場合は、用語としては「Polyfill」が一般的かなと思いました
https://developer.mozilla.org/ja/docs/Glossary/Polyfill

</details>

<details>
  <summary>製品利用することはできますか？</summary>
  <p><a href="https://vercel.com">https://vercel.com</a> は当初から Next.js で実装されています。</p>

  <p>開発経験とエンドユーザーのパフォーマンスの双方にとって満足いくものになったと考えたため、コミュニティに共有することを決めました。</p>
</details>

<details>
  <summary>ファイルサイズはどれくらいですか？</summary>

>   <p>クライアントサイドのバンドルサイズはアプリごとに計測されるべきですが、 Next のメインバンドルは gzip 形式で約 65kb に圧縮されています。</p>

Reviewer: ```suggestion
  <p>クライアントサイドのバンドルサイズはアプリごとに計測されるべきですが、軽量の Next.js アプリであればメインバンドルは gzip 形式で約 65kb に圧縮されています。</p>
```

small という部分の訳が抜けていたので追加しました！

</details>

<details>
  <summary>webpack の内部設定を変えるにはどうすればいいですか？</summary>

>   <p>Next.js は webpack 設定のオーバーヘッドを無くすように最善を尽くしています。 しかし、独自で設定する必要がある場合は、こちらの <a href="/docs/api-reference/next.config.js/custom-webpack-config.md">webpack 設定のカスタマイズのドキュメント</a>を見てください。</p>

Reviewer: ```suggestion
  <p>Next.js は webpack 設定のオーバーヘッドを無くすように最善を尽くしています。 しかし、より多くの制御を必要とするような高度なケースでは、こちらの <a href="/docs/api-reference/next.config.js/custom-webpack-config.md">webpack 設定のカスタマイズのドキュメント</a>を見てください。</p>
```

正しい訳だとは思うのですが、より原文に忠実になるように修正してみました。

>   <p>Next.js は webpack 設定のオーバーヘッドを無くすように最善を尽くしてきました。 もし独自で設定する必要がある場合は、こちらの <a href="/docs/api-reference/next.config.js/custom-webpack-config.md">webpack 設定のカスタマイズのドキュメント</a>を見てください。</p>

Reviewer 2: ```suggestion
  <p>Next.js は webpack 設定のオーバーヘッドを無くすように最善を尽くしています。 しかし、独自で設定する必要がある場合は、こちらの <a href="/docs/api-reference/next.config.js/custom-webpack-config.md">webpack 設定のカスタマイズのドキュメント</a>を見てください。</p>
```
過去形の表現となっていたので今も最善を尽くしてるというニュアンスが伝わるようにしてみました

</details>

<details>
  <summary>コンパイルされている構文機能は何ですか？ それを変更するにはどうしたらいいですか？</summary>
  <p>V8 を踏襲しています。 V8 は ES6 と async/await をサポートしてきたので Next.js にもコンパイルされています。 V8 はクラスデコレーターをサポートしていないので Next.js でもコンパイルされていません。</p>

  <p>もっと知りたい場合は <a href="/docs/advanced-features/customizing-babel-config.md">babel 設定のカスタマイズ</a>のドキュメントを見てください。</p>
</details>

<details>
  <summary>なぜ新しいルーターなのですか？</summary>
  Next.js には以下の特徴があります:
  <ul>

>     <li>ルートは事前に知られておく必要はなく、ルートマニフェストを送りません</li>

Reviewer 2: ```suggestion
    <li>ルーティングは事前に知られておく必要はなく、ルートマニフェストを送りません</li>
```
`Routes` は違和感なければ「ルーティング」と訳してしまったほうが自然かなと思いました

>     <li>ルートは常に遅延読み込みが可能です</li>

Reviewer 2: ```suggestion
    <li>ルーティングは常に遅延読み込みが可能です</li>
```

  </ul>
</details>

<details>
  <summary>データを取得するにはどうすればいいですか？</summary>

>   <p>好きな方法を選ぶことができます。 React コンポーネント内でリモートデータを取得するには、 <a href="https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch">fetch API</a> か <a href="https://swr.now.sh/">SWR</a> のどちらかを使うことができます。 もしくは Next.js 独自の<a href="/docs/basic-features/data-fetching.md">データ取得メソッド</a>を使って初期データ群を取得することができます。</p>

Reviewer: ```suggestion
  <p>好きな方法を選ぶことができます。 React コンポーネント内でリモートデータを取得するには、 <a href="https://developer.mozilla.org/ja/docs/Web/API/Fetch_API/Using_Fetch">fetch API</a> か <a href="https://swr.now.sh/">SWR</a> のどちらかを使うことができます。 もしくは Next.js 独自の<a href="/docs/basic-features/data-fetching.md">データ取得メソッド</a>を使って初期データを追加できます。</p>
```

- mozilla のドキュメントは日本語verがあったのでそちらに URL を修正しました。
- initial data population の訳は [こちら](https://github.com/Nextjs-ja-translation/Nextjs-ja-translation-docs/pull/121/#discussion_r460481230) と平仄を取り、「初期データの追加」としました。

</details>

<details>
  <summary>GraphQL と一緒に使えますか？</summary>

>   <p>使えます！ <a href="https://github.com/zeit/next.js/tree/canary/examples/with-apollo">example with Apollo</a> を見てください。</p>

Reviewer 2: ```suggestion
  <p>使えます！ <a href="https://github.com/vercel/next.js/tree/canary/examples/with-apollo">example with Apollo</a> を見てください。</p>
```

>   <p>使えます！ <a href="https://github.com/vercel/next.js/tree/canary/examples/with-apollo">example with Apollo</a> を見てください。</p>

Reviewer: ```suggestion
  <p>使えます！ <a href="https://github.com/vercel/next.js/tree/canary/examples/with-apollo">Apollo を使ったサンプル</a> を見てください。</p>
```

example with Apollo の部分を日本語にしました。

</details>

<details>
  <summary>Reduxと一緒に使えますか？</summary>

>   <p>使えます！ <a href="https://github.com/zeit/next.js/tree/canary/examples/with-redux">example</a> を見てください。 <a href="https://github.com/zeit/next.js/tree/canary/examples/with-redux-thunk">example with thunk</a> もあります。</p>

Reviewer 2: ```suggestion
  <p>使えます！ <a href="https://github.com/zeit/next.js/tree/canary/examples/with-redux">example</a> を見てください。 <a href="https://github.com/vercel/next.js/tree/canary/examples/with-redux-thunk">example with thunk</a> もあります。</p>
```

>   <p>使えます！ <a href="https://github.com/vercel/next.js/tree/canary/examples/with-redux">example</a> を見てください。 <a href="https://github.com/vercel/next.js/tree/canary/examples/with-redux-thunk">example with thunk</a> もあります。</p>

Reviewer: ```suggestion
  <p>使えます！ こちらの<a href="https://github.com/vercel/next.js/tree/canary/examples/with-redux">サンプル</a> を見てください。 <a href="https://github.com/vercel/next.js/tree/canary/examples/with-redux-thunk">thunk を用いたサンプル</a> もあります。</p>
```

example と example with thunk の部分を日本語にしました。

</details>

<details>
  <summary>静的ファイルのために CDN を使うことはできますか？</summary>
  <p>はい。<a href="/docs/api-reference/next.config.js/cdn-support-with-asset-prefix.md">こちら</a>に詳しく書いてあります。</p>
</details>

<details>
  <summary>Next でお気に入りの JavaScript ライブラリやツールを使うことはできますか？</summary>

>   <p>最初のリリース以来、多くの他ライブラリやツールとの併用例がコントリビュートされてきました。 <a href="https://github.com/zeit/next.js/tree/canary/examples">examples</a> ディレクトリで見ることができます。</p>

Reviewer 2: ```suggestion
  <p>最初のリリース以来、多くの他ライブラリやツールとの併用例がコントリビュートされてきました。 <a href="https://github.com/vercel/next.js/tree/canary/examples">examples</a> ディレクトリで見ることができます。</p>
```

</details>

<details>
  <summary>何から影響を受けて開発されましたか？</summary>
  <p>私たちが設定した達成目標の多くは、 Guillermo Rauch の <a href="https://rauchg.com/2014/7-principles-of-rich-web-applications">7 principles of Rich Web Applications</a> に記されているものです。</p>

  <p>PHP の使いやすさから大きな影響を受けています。 HTML を出力するために PHP を使わなければならない多くの場合において、 Next.js は最適な代替手段になると思っています。</p>

>   <p>PHP とは違って ES6 モジュールシステムの恩恵を受けることができ、ページ毎にコンポーネントや関数がエクスポートされることで遅延評価やテストを簡単に導入することができます。</p>

Reviewer: ```suggestion
  <p>PHP とは違って ES6 モジュールシステムの恩恵を受けることができ、ページ毎にコンポーネントや関数がエクスポートされており、遅延評価やテストのためにそれらを簡単にインポートできます。</p>
```

原文そのものの意味は「各ページからコンポーネントや関数がエクスポートされているから、遅延評価やテストにおいてそれら（コンポーネントや関数）を簡単にインポートできる」で、「遅延評価やテストを簡単に導入できる」という意味とは少し異なるかなと思いましたので、その部分を修正しました。

>   <p>手間がかからない React のサーバーサイドレンダリングの選択肢を調査していた時、 Next.js と同じ手法を取っている <a href="https://github.com/facebookarchive/react-page">react-page</a> （現在は非推奨）に出会いました。これは React の製作者である Jordan Walke によるものです。</p>

Reviewer 2: ```suggestion
  <p>手間がかからない React のサーバーサイドレンダリングの選択肢を調査していた時、 Next.js と似た手法を取っている <a href="https://github.com/facebookarchive/react-page">react-page</a> （現在は非推奨）に出会いました。これは React の製作者である Jordan Walke によるものです。</p>
```
`similar` なので「同じ」よりも「似た」という表現にとどめておく方が適切かなと思いました
